oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/UserAgentImpl.java

```diff
@@ -116,6 +144,15 @@ AuthorizationResponse encode(final String methodName, final String... parameters
         }
     }
 
+    static void relayProperties(final Properties properties, final Set<String> propertyNames, final List<String> destinationCommand) {
+        for (final String propertyName : propertyNames) {
+            final String propertyValue = properties.getProperty(propertyName);
+            if (propertyValue != null) {
+                destinationCommand.add("-D" + propertyName + "=" + propertyValue);

```

Comment:
> I am still unclear about how to work with a proxy that requires authentication.

You're correct: I did not test that scenario.  The [Networking Properties](http://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html) page seems to indicate there's no way to configure HTTP and HTTPS proxy servers for authentication using system properties.

It's possible the `java.net.useSystemProxies` property would enable this transparently such that the proxy credentials wouldn't have to be visible in the configuration file as well as in process command-line arguments.

I've logged issue #18 to track this.